scripts/deploy.sh

```diff
@@ -0,0 +1,223 @@
+for tag in $EXISTING_TAGS; do
+    if ! version_gt "$VERSION" "$tag"; then
+        echo -e "${RED}Error: Version $VERSION is not greater than existing version $tag${NC}"
+        exit 1
+    fi
+done
+
+# Configuration
+DEPLOYMENT_REPO="deployment-wallet"
+DEPLOY_SERVERS=("deploy_wallet@web-1" "deploy_wallet@web-2" "deploy_wallet@web-3" "deploy_wallet@web-4")
+
+# Set environment tag based on BUILD_ENV
+ENV_TAG=$([ "$BUILD_ENV" = "mainnet" ] && echo "main" || echo "test")
+
+# Pre-deployment tasks
+echo -e "${BLUE}Running pre-deployment tasks...${NC}"
+
+if [ "$SYNC_TRANSLATIONS" = "true" ]; then
+    echo -e "${CYAN}Syncing translations...${NC}"
+    yarn i18n:sync
+fi
+
+# Get up-to-date EBA RT1 bank list (disabled while OASIS is not available)
+# yarn utils:getBankList
+
+# Get up-to-date browsers support list from caniuse.
+# The browserslist utility is meant to run via npx, even when usually using yarn, and is compatible with yarn.lock.
+echo "Updating browsers support list..."
+npx browserslist@latest --update-db
+
+# Build Nginx path allowlist
+echo "Building Nginx path allowlist..."
+yarn utils:makeNginxAllowlist
+
+# Check for uncommitted changes
+if [[ `git status --porcelain` ]]; then
+    echo -e "${RED}ERROR: The repository has uncommitted changes. Commit them first, then run again.${NC}"
+    exit 1
+fi
+
+# Function to create commit/tag message
+create_message() {
+    echo "Nimiq Wallet v$VERSION
+
+$COMMIT_MSG
+
+$EXCLUDE_RELEASE"
+}
+
+# Create and push source tag
+echo -e "${BLUE}Creating source tag v$VERSION...${NC}"
+git tag -a -s "v$VERSION" -m "$(create_message)"
+
+# Build the project
+echo -e "${BLUE}Building project with $BUILD_ENV configuration...${NC}"
+env "build=$BUILD_ENV" yarn build
+
+# Push changes and tags
+echo -e "${BLUE}Pushing source changes and tags...${NC}"
+git push && git push --tags
+
+# Deploy to deployment repository
+echo -e "${BLUE}Deploying to $DEPLOYMENT_REPO...${NC}"
+cd "$DEPLOYMENT_REPO" || exit 1
+
+# Checkout appropriate branch and pull latest changes
+DEPLOY_BRANCH=$([ "$BUILD_ENV" = "mainnet" ] && echo "mainnet" || echo "testnet")
+echo -e "${CYAN}Checking out $DEPLOY_BRANCH branch...${NC}"
+git checkout "$DEPLOY_BRANCH"
+git pull
+
+# Check if new version is greater than all existing tags in deployment repo for the current branch
+echo -e "${BLUE}Checking version against existing tags in deployment repo ($DEPLOY_BRANCH branch)...${NC}"
+EXISTING_DEPLOY_TAGS=$(git tag | grep "^v[0-9].*-$ENV_TAG-" | sed 's/^v\([0-9][^-]*\).*/\1/')
+for tag in $EXISTING_DEPLOY_TAGS; do
+    if ! version_gt "$VERSION" "$tag"; then
+        echo -e "${RED}Error: Version $VERSION is not greater than existing version $tag in deployment repo${NC}"
+        exit 1
+    fi
+done
+
+# Copy build files
+echo -e "${CYAN}Copying build files...${NC}"
+cp -r ../dist/. dist
+git add dist
+
+# Commit changes
+echo -e "${CYAN}Committing changes...${NC}"
+git commit -m "$(create_message)"
```

Comment:
I personally always do a `git status` after adding and before commiting, to check if it doesn't look totally off what I'm doing. E.g. if there's only a minor change in the source, I don't expect all files to have been replaced and vice versa.

So I'd like a confirm step here.

Reply:
Make sense, will add that then 👍